src/executor/current_thread/mod.rs

```diff
@@ -569,9 +590,41 @@ impl<'a, P: Park> Entered<'a, P> {
 
     /// Returns `true` if any futures were processed
     fn tick(&mut self) -> bool {
-        self.executor.scheduler.tick(
+        // Spawn any futures that were spawned from other threads by manually
+        // looping over the receiver stream
+
+        // FIXME: Slightly ugly but needed to make the borrow checker happy
+        let (mut borrow, spawn_receiver) = (
+            Borrow {
+                scheduler: &mut self.executor.scheduler,
+                num_futures: &mut self.executor.num_futures,
+            },
+            &mut self.executor.spawn_receiver,
+        );
+
+        let notify = borrow.scheduler.notify();
+
+        loop {
+            let res = borrow.enter(self.enter, || {
```

Comment:
A lot of this can probably be simplified by not using the future enabled channel.

Reply:
Yes